src/unix/pty.cc

```diff
@@ -229,98 +355,54 @@ NAN_METHOD(PtyFork) {
   int uid = info[6]->IntegerValue(Nan::GetCurrentContext()).FromJust();
   int gid = info[7]->IntegerValue(Nan::GetCurrentContext()).FromJust();
 
-  // fork the pty
   int master = -1;
+  pid_t pid;
+  int error;
 
-  sigset_t newmask, oldmask;
-  struct sigaction sig_action;
-
-  // temporarily block all signals
-  // this is needed due to a race condition in openpty
-  // and to avoid running signal handlers in the child
-  // before exec* happened
-  sigfillset(&newmask);
-  pthread_sigmask(SIG_SETMASK, &newmask, &oldmask);
+  if (strlen(cwd) || uid != -1 || gid != -1) {
```

Comment:
Would `posix_spawnattr_set_uid_np` `posix_spawnattr_set_gid_np` help here to also handle the gid/uid case, as mentioned by @deepak1556?

Reply:
Added both - but I'm getting `warning: weak identifier 'posix_spawnattr_set_uid_np' never declared`, but none for `posix_spawn_file_actions_addchdir_np`. I suspect this is specific to my system, but would appreciate someone taking a look at it